libs/application/template-api-modules/src/lib/modules/templates/european-health-insurance-card/european-health-insurance-card.service.ts

```diff
@@ -0,0 +1,229 @@
+        ?.childrenCustodyInformation as unknown) as NationalRegistry[]
+      for (let i = 0; i < custodyData?.length; i++) {
+        nridArr.push(custodyData[i].nationalId)
+      }
+      return nridArr
+    }
+    return application.answers[applyType] as string[]
+  }
+
+  toCommaDelimitedList(arr: string[]) {
+    let listString = ''
+    for (let i = 0; i < arr.length; i++) {
+      listString += arr[i]
+      if (i !== arr.length - 1) {
+        listString += ','
+      }
+    }
+    return listString
+  }
+
+  /** Get's applicants that want a temporary card (PDF) and their current physical (plastic) card number */
+  getPDFApplicantsAndCardNumber(
+    application: ApplicationWithAttachments,
+  ): ApplicantCard[] {
+    const pdfApplicantArr: ApplicantCard[] = []
+
+    const applicants = this.getApplicants(
+      application,
+      FormApplyType.APPLYING_FOR_PDF,
+    )
+
+    // Initial card Response
+    const cardResponse = application.externalData.cardResponse
+      ?.data as CardResponse[]
+    // New card response
+    const newPlasticCardsResponse = application.externalData
+      .applyForCardsResponse?.data as CardResponse[]
+
+    if (applicants) {
+      for (let i = 0; i < applicants.length; i++) {
+        if (newPlasticCardsResponse && newPlasticCardsResponse.length > 0) {
+          const newCardResponse = newPlasticCardsResponse.find(
+            (x) => x.applicantNationalId === applicants[i],
+          )
+          if (newCardResponse) {
+            const plasticCard = newCardResponse?.cards?.find(
+              (x) => x.isPlastic === true,
+            )
+            if (plasticCard) {
+              pdfApplicantArr.push({
+                cardNumber: plasticCard.cardNumber ?? '',
+                nationalId: applicants[i],
+              })
+              continue
+            }
+          }
+        }
+
+        const currentCardResponse = cardResponse.find(
+          (x) => x.applicantNationalId === applicants[i],
+        )
+        if (currentCardResponse) {
+          const plasticCard = currentCardResponse?.cards?.find(
+            (x) => x.isPlastic === true,
+          )
+          if (plasticCard) {
+            pdfApplicantArr.push({
+              cardNumber: plasticCard?.cardNumber ?? '',
+              nationalId: applicants[i],
+            })
+          }
+        }
+      }
+    }
+
+    return pdfApplicantArr
+  }
+
+  async getCardResponse({ auth, application }: TemplateApiModuleActionProps) {
+    const nridArr = this.getApplicants(application)
+
+    try {
+      const resp = await this.ehicApi
+        .withMiddleware(new AuthMiddleware(auth as Auth))
+        .cardStatus({
+          applicantnationalids: this.toCommaDelimitedList(nridArr),
+        })
+
+      if (!resp) {
+        this.logger.error('EHIC.API response empty from getCardResponse', resp)
+      }
+
+      return resp
+    } catch (e) {
+      this.logger.error('EHIC.API error getCardResponse', e)
```

Comment:
FYI  these error logs are redundant. The template api catches all errors and logs them out with the both the templateId and the actionId. see [here](https://github.com/island-is/island.is/blob/main/libs/application/template-api-modules/src/lib/modules/template-api.service.ts#L65)